src/popover/container.tsx

```diff
@@ -119,11 +122,15 @@ export default function PopoverContainer({
       const shouldKeepPosition = keepPosition && onContentResize && !!previousInternalPositionRef.current;
       const fixedInternalPosition = (shouldKeepPosition && previousInternalPositionRef.current) ?? undefined;
 
+      // On mobile screens, allow the popover to open outside of the viewoprt and scroll to it,
+      // if there is no position where it can fit without being cropped.
+      const allowVerticalOverflow = isMobile && !shouldKeepPosition;
```

Comment:
`isMobile` depends on the window _width_. Vertical overflow happens when the window _height_ is too small. 

Should we rely on a different condition to make overflow happening?

Reply:
The reason to distinguish between mobile and desktop is to avoid the scrolling while hovering, so I'll try something else somehow based on hover vs tap rather than on the screen size.

Reply:
What does it have to do with device type? If the bug says „popover does not fit vertically“ - the fix probably should count on vertical size. If anybody has a 1000x200 window for some reason, the fix should work for them too

Reply:
The device type distinction comes from two reasons:
- On desktop chart popovers are opened on hover, and triggering auto-scroll on hover can lead to a jarring experience
- A scrollable popover on mobile is more concerning than on desktop because there are no scrollbars that indicate its scrollability, so it's preferable to auto-scroll the window instead

Also considering a use case like the one depicted [here](https://github.com/cloudscape-design/components/pull/285), where chaning the behavior seems unnecessary. Although for such a case, setting a minimum height to the popover before going for auto-scroll vs internal popover scroll would also work.

Reply:
I think I got what you wanted to do here. You want to apply fix only when popover opens by tapping on it, skipping the hover interactions. 

If this is what you want, we can dive into solutions for this

Reply:
> A scrollable popover on mobile is more concerning than on desktop because there are no scrollbars

There are no scrollbars on macs, big and small. If this is a factor you find important for the fix, it should be cross-platform.

----

As for the conditions to activate this behavior, this sounds like it should be activated when `isPinned` flag on chart-popover is active. What do you think?

Reply:
> There are no scrollbars on Apple devices, big and small. If this is a factor you find important for the fix, it should be cross-platform.

Actually to be precise there are no scrollbars on Apple devices (by default) _when using touch interfaces_ (mobile or trackpad), so I am more inclined to be specific to tap vs click as you suggested earlier. Using `isPinned` could lead to undesired resizing flows when pinning for popovers that were already open on hover (i.e, chart popovers).

I guess instead of
> A scrollable popover on mobile is more concerning than on desktop because there are no scrollbars that indicate its scrollability

to be more precise I should have said
> A scrollable popover on Apple devices with touch interface is more concerning than on desktop because there are no scrollbars that indicate its scrollability (by default)

PS: To check that "Automatically based on mouse or trackpad" is the default option, you can run `defaults delete -g AppleShowScrollBars` and reopen the Appearance settings.

Reply:
Parked for offline discussion. I will get back to reviewing this after the offline discussion happens